llm_unlearn_ucl/unlearn_harm.py

```diff
@@ -229,25 +237,47 @@ def main(args) -> None:
     assert (
         args.samples_count // args.sequential
     ) % args.batch_size == 0, "samples in each 'sequence' (--samples_count / --sequential) should be a multiple of batch_size."
-    accelerator = Accelerator(
-        mixed_precision="bf16"
-    )  # accelerator precision can be specified if required.
+
+    if args.wandb_log:
+        accelerator = Accelerator(log_with="wandb", mixed_precision="bf16")
+        accelerator.init_trackers(
+            project_name=args.wandb_project_name,
+            config=vars(args),
+            init_kwargs={"wandb": {"name": args.wandb_run_name}},
+        )
+    else:
+        accelerator = Accelerator(mixed_precision="bf16")
     device = accelerator.device
 
+    # setup logging
+    logging.basicConfig(
+        filename=args.log_file,
+        filemode="w+",
+        format="%(asctime)s,%(msecs)d %(name)s %(levelname)s %(message)s",
+        datefmt="%Y-%m-%d-%H-%M",
+        level=logging.INFO,
+    )
+    logger = get_logger(__name__)
+    if accelerator.is_main_process:
+        for arg in vars(args):
+            logger.info(f"{arg}: {getattr(args, arg)}")
+    accelerator.wait_for_everyone()
+
     print(f"Loading model {args.model_name} for training...")
     if args.use_quantized:
         # Uncomment for quantized
         model = AutoModelForCausalLM.from_pretrained(
             args.model_name,
             cache_dir=args.cache_dir,
             load_in_8bit=True,
-            torch_dtype=torch.float32,
+            torch_dtype=torch.bfloat16,
```

Comment:
For visibility: @Willmish's code removes bf16 from the python code and defines it in the deepspeed config.